packages/@lwc/engine/dom/src/apis/create-element.ts

```diff
@@ -116,6 +116,7 @@ export function createElement(
     setElementProto(element, def);
 
     createVM(element, def, {
+        tagName: sel,
```

Comment:
I'm concern that if someone is doing `createElement('x-Foo', { is: Foo })`, it will ended being an invalid tagName here. For WC we extract it from the element and lowercase it; when from the compiled template, it is probably safer, and always coming as lowercase, but our own createElement do not lowercase it.

Reply:
cc @ekashida

Reply:
Discussed offline and agreed to leave this as-is for now since we couldn't think of a scenario where it would make a difference.